packages/adblocker-webextension/src/index.ts

```diff
@@ -128,19 +139,98 @@ export function updateResponseHeadersWithCSP(
   return { responseHeaders };
 }
 
+const HTML_FILTERABLE_MIME_TYPES = new Set([
+  'application/javascript',
+  'application/json',
+  'application/mpegurl',
+  'application/vnd.api+json',
+  'application/vnd.apple.mpegurl',
+  'application/vnd.apple.mpegurl.audio',
+  'application/x-javascript',
+  'application/x-mpegurl',
+  'application/xhtml+xml',
+  'application/xml',
+  'audio/mpegurl',
+  'audio/x-mpegurl',
+]);
+export const MAXIMUM_RESPONSE_BUFFER_SIZE = 10 * 1024 * 1024;
+
+function getHeaderFromDetails(
+  details: OnHeadersReceivedDetailsType,
+  headerName: WebRequest.HttpHeadersItemType['name'],
+): string | undefined {
+  return details.responseHeaders?.find((header) => header.name.toLowerCase() === headerName)
+    ?.value;
+}
+
+export function shouldApplyReplaceSelectors(
+  request: Request<OnBeforeRequestDetailsType | OnHeadersReceivedDetailsType>,
+): boolean {
+  const details = request._originalRequestDetails as OnHeadersReceivedDetailsType;
+
+  // In case of undefined error of xhr/fetch and any kind of network activities
+  if (details.statusCode === 0) {
+    return false;
+  }
+
+  // If status code is non 2xx or 4xx, 5xx
+  const statusCodeScope = (details.statusCode / 100) >>> 0;
+  if (statusCodeScope !== 2 && statusCodeScope < 4) {
+    return false;
+  }
```

Comment:
Don't we want to skip for any non-2xx code?
```suggestion
  // If status code is non 2xx or 4xx, 5xx
  if (details.statusCode < 200 || details.statusCode > 299) {
    return false;
  }
```